docs: update keywords.json

allieorth opened the pull request:
**Related Issue:** https://github.com/ArcGIS/icon-explorer/issues/6

## Summary
Updated the keywords.json file by merging codepoint values from fantasticonrc.json.
Each icon entry now includes a new "codepoint" field, populated from the codepoints list or set to an empty string if missing.
This is in the hope that we can update the reference page to Supports improved search and integration for the icon font, enabling codepoint-based lookups based on keywords.

allieorth committed:
replaced keywords.json file

added new version containing tag for codepoints to make font searchable.

allieorth committed:
launched ref page

jcfranco reviewed (changes requested):
Can you revert the changes to the lockfile?

allieorth commented:
> Can you revert the changes to the lockfile?

Sure — just a heads-up: that change happened automatically when I ran the launch command, which we do for every icon release. If I revert it now, it’ll likely reappear with the next icon release.
Is there a specific issue you’re seeing occur with it?

jcfranco commented:
> > Can you revert the changes to the lockfile?
> 
> Sure — just a heads-up: that change happened automatically when I ran the launch command, which we do for every icon release. If I revert it now, it’ll likely reappear with the next icon release. Is there a specific issue you’re seeing occur with it?

It looks like `launch` runs `npm install`, which can update the lockfile depending on the machine, but those changes shouldn’t be included when updating `keywords.json` as they are not related.

allieorth committed:
Revert "launched ref page"

This reverts commit 335288ddd7394bf45e603a3d50e4bd9016819549.

allieorth committed:
Merge branch 'dev' into allieorth/update-ui-icons-keywords

jcfranco commented on packages/ui-icons/docs/keywords.json:
```diff
@@ -10112,57 +11454,68 @@
     "category": "Generic",
-    "release": "4.4.0"
+    "release": "4.4.0",
+    "codepoint": 61836"
```
The trailing double-quote is invalid here.

jcfranco commented on packages/ui-icons/docs/keywords.json:
```diff
@@ -5841,7 +6613,8 @@
     "category": "Generic",
-    "release": "3.19.1"
+    "release": "3.19.1",
+    "codepoint": ""
```
Might be better to use `null` for the ones that don't have a codepoint value. Otherwise, we're mixing strings and numbers, whereas `null` represents no value. Applies to all 'empty' `codepoint` fields.

jcfranco commented on packages/ui-icons/docs/keywords.json:
```diff
@@ -2442,50 +2825,58 @@
       "...",
       "dot dot dot",
-      "…",
+      "\u2026",
```
Is this change expected? Applies to other `...` entries.

allieorth replied:
Nope this isnt expected — let me look into why this is happening

jcfranco commented on packages/ui-icons/docs/keywords.json:
```diff
@@ -4278,77 +4902,92 @@
   },
   "effects": {
-    "alias": ["3.16.3", "objects", "flash", "sparkles", "filters", "stars", "constellations", "flashy", "shiny"],
+    "alias": ["3.27.4", "objects", "ai", "sparkle", "system", "machine learning"],
```
Are these additional metadata changes intentional?

jcfranco commented on packages/ui-icons/docs/keywords.json:
```diff
@@ -8396,142 +9460,164 @@
     "category": "Lists",
-    "release": "3.27.0"
-  },
-  "effects": {
```
It looks like this got merged into the previous `"effects"` item.

jcfranco commented on packages/ui-icons/docs/keywords.json:
```diff
@@ -9621,427 +10875,506 @@
     "category": "Generic",
-    "release": "4.2.1"
-  },
-  "spike-data-visualization-outlined-open": {
```
Looks like this item got merged with the previous `"spike-data-visualization-outlined-open"`.

jcfranco commented on packages/ui-icons/docs/keywords.json:
```diff
@@ -10112,57 +11454,68 @@
     "category": "List",
-    "release": "4.4.0"
+    "release": "4.4.0",
+    "codepoint": 61834
```
If there's a script for keeping these in sync, we should add it and include in the build.

jcfranco replied:
Per our conversation, the codepoint syncing script will be set up in a separate PR.

allieorth replied:
Great! Yup I'll follow this up with a separate PR.
Thanks ✨

allieorth committed:
updated keywords.json file

fixed some unexpected changes and errors
upodated `codepoint: ""` to now be `codepoint:null`
resolved duplicate tags for spike-data-visualization-outlined-open and effects

allieorth committed:
added script for syncing the codepoints file with the keywords file

jcfranco reviewed (approved):
🚀

allieorth committed:
Revert "added script for syncing the codepoints file with the keywords file"

This reverts commit 384a8a4c53db7e58894f53bae575d3b1826efcfb.

macandcheese commented:
Looks good to me, once merged and released I can pull in to the icon browsing reference site.

allieorth committed:
Updated to include new icon tag and eliminated duplicate entries

Added new tags and codepoint for `file-esl`
Eliminated duplicate entries for `spike-data-visualization-outlined-open` and `effects` icons and combined tags.

allieorth committed:
Merge branch 'dev' into allieorth/update-ui-icons-keywords
